File: mater-webapp/src/app/store/dtos/personal.dto.ts
import {AbstractDto} from "./abstract.dto";

export class PersonalDto extends AbstractDto {
  firstName: string;
  firstNameSet: boolean;
  lastName: string;
  lastNameSet: boolean;
  middleName: string;
  middleNameSet: boolean;
  birthDate: Date;
  birthDateSet: boolean;
}

File: mater-rest/src/main/java/com/oneune/mater/rest/main/store/dtos/PersonalDto.java
package com.oneune.mater.rest.main.store.dtos;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;

/**
 * @see com.oneune.mater.rest.main.store.entities.PersonalEntity
 */
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Data
public class PersonalDto extends AbstractDto {
    String firstName;
    boolean firstNameSet;
    String lastName;
    boolean lastNameSet;
    String middleName;
    boolean middleNameSet;
    LocalDate birthDate;
    boolean birthDateSet;
}
